### PR TITLE
Fix jingle action

### PR DESCRIPTION
**Pull request by matteocampana**

### CODE REVIEW

File: smack-jingle-old/src/main/java/org/jivesoftware/smackx/jingleold/packet/Jingle.java [hunk 1]
```diff
@@ -354,7 +354,7 @@ protected IQChildElementXmlStringBuilder getIQChildElementBuilder(IQChildElement
             buf.append(" responder=\"").append(getResponder()).append('"');
         }
         if (getAction() != null) {
-            buf.append(" action=\"").append(getAction().name()).append('"');
+            buf.append(" action=\"").append(getAction().toString()).append('"');
```
Comment (Flowdalic) [inline smack-jingle-old/src/main/java/org/jivesoftware/smackx/jingleold/packet/Jingle.java]: I'm missing an explanation regarding what is wrong about using `name()` here?

Comment (vanitasvitae) [reply smack-jingle-old/src/main/java/org/jivesoftware/smackx/jingleold/packet/Jingle.java]: I'd assume that since `JingleActionEnum` takes a String value in its constructor, `action.name()` would return for example `CONTENT_ACCEPT` which is the name of the enum, while `action.toString()` will return the value of actionCode (in this case `content-accept`), which is what you actually want to put into the XML.

Comment (matteocampana) [reply smack-jingle-old/src/main/java/org/jivesoftware/smackx/jingleold/packet/Jingle.java]: Yes, exactly. 
And because of that, the parser sets `action=null` due to `JingleActionEnum` `getActiongetAction(String inActionCode)` method.
See also this discussion: https://discourse.igniterealtime.org/t/null-pointer-exception-in-jinglemanager/73226/5

File: smack-core/src/test/java/org/jivesoftware/smack/test/util/CharsequenceEquals.java [hunk 2]
```diff
@@ -1,47 +0,0 @@
-/**
```
Comment (vanitasvitae) [inline smack-core/src/test/java/org/jivesoftware/smack/test/util/CharsequenceEquals.java]: Can you revert this change, as it is already taken care for in #209?

Comment (matteocampana) [reply smack-core/src/test/java/org/jivesoftware/smack/test/util/CharsequenceEquals.java]: Yes sure, I will do it.

**Comment by vanitasvitae** [comment]
Can you please hard reset your branch to master, only apply the changes from 7cb9175 and force push to get rid of the other commits?

**Comment by Flowdalic** [comment]
Picked onto 4.2 as https://github.com/igniterealtime/Smack/commit/713f0438df49c83da400975a054d384701da6e16
